src/exchanges.js

```diff
@@ -35,10 +21,16 @@ var Publisher = function(conn, channel, entries, exchangePrefix, options) {
   this._channel = channel;
   this._entries = entries;
   this._options = options;
-  if (options.drain) {
-    assert(options.component, "component name for statistics is required");
-    assert(options.process,   "process name for statistics is required");
-    this._reporter = ExchangeReports.reporter(options.drain);
+  if (options.drain || options.component) {
+    console.log('taskcluster-lib-stats is now deprecated!\n' +
+                'Use the `monitor` option rather than `drain`.\n' +
+                '`monitor` should be an instance of taskcluster-lib-monitor.\n' +
+                '`component` is no longer needed. Prefix your `monitor` before use.');
+  }
+
+  var monitor = null;
+  if (options.monitor) {
+    monitor = options.monitor.prefix('pulse-publisher');
```

Comment:
like `api` shouldn't this be done externally? Just in case someone decides to have two exchange declarations in the same component... also wouldn't `exchanges` be a better prefix (or something else shorter).. Maybe just `publisher`...